Assets/Scripts/Combat/Combantants/Combatant.cs

```diff
@@ -14,16 +14,31 @@ public abstract class Combatant : MonoBehaviour
     public int CurrentHealthPoints;
     public int MaxShieldPoints;
     public int CurrentShieldPoints;
-    private readonly float HealthBarPosition = 125f;
     protected CombatSystem CombatSystem;
     [SerializeField]
     GameObject HealthBarPrefab;
     [SerializeField]
+    GameObject ShieldBarPrefab;
+    [SerializeField]
     GameObject HealthBarUIPanel;
     [SerializeField]
     GameObject Shield;
 
     private GameObject HealthBar;
+    private HealthBar RedBar;
```

Comment:
Why "RedBar" and "BlueBar"? Also, seems redundant to have both a reference to the ` private GameObject HealthBar;` AND its `private HealthBar RedBar;`. Just use the `HealthBar`, and call GetComponent ---- or vice versa

Reply:
Doing `HealthBar.GetComponentInChildren<PointsBar>().MaxValue = MaxHealthPoints` every frame seems counter intuitive, having a reference to it is easier to read and would take out the GetComponentInChildren<> overhead

Reply:
Sure!

As a heads up, be aware that this introduces overhead for maintaining both refs. Maybe not for this circumstance, but in other scripts, if one ref B changed where it's pointing and you forget to update A, then you can run into bugs.